File: src/main/java/dev/reinaldosantos/car_fix/dto/LoginDto.java
package dev.reinaldosantos.car_fix.dto;

import dev.reinaldosantos.car_fix.customAnnotations.PasswordValidator;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class LoginDto {
    @NotNull(message = "Email is required")
    @Email(message = "Please enter valid email")
    private String email;
    @PasswordValidator
    private String password;
    @NotNull(message = "Please enter token phone")
    private String tokenPhone;

}

File: src/dto/login.dto.ts
/* eslint-disable indent */
import { IsEmail, IsString } from "class-validator";

export class LoginDto {
    @IsEmail({}, { message: "Please enter valid email" })
    email: string;
    @IsString({ message: "Please enter valid password" })
    password: string;
    @IsString({ message: "Please enter token" })
    token_phone?: string;
    @IsString({ message: "Please enter to type" })
    type: string;
    constructor(
        email: string,
        password: string,
        type: string,
        token_phone?: string,
    ) {
        this.email = email;
        this.password = password;
        this.token_phone = token_phone;
        this.type = type;
    }
}